specification/sections/04-hooks.md

```diff
@@ -129,15 +132,17 @@ Evaluation context merge order is defined in [Context levels and merging](./03-e
 
 #### Requirement 4.3.6
 
-> The `after` stage **MUST** run after flag resolution occurs. It accepts a `hook context` (required), `flag evaluation details` (required) and `hook hints` (optional). It has no return value.
+> The `after` stage **MUST** run after flag resolution occurs. It accepts a `hook context` (required), `evaluation details` (required) and `hook hints` (optional). It has no return value.
 
 #### Requirement 4.3.7
 
 > The `error` hook **MUST** run when errors are encountered in the `before` stage, the `after` stage or during flag resolution. It accepts `hook context` (required), `exception` representing what went wrong (required), and `hook hints` (optional). It has no return value.
 
 #### Requirement 4.3.8
 
-> The `finally` hook **MUST** run after the `before`, `after`, and `error` stages. It accepts a `hook context` (required) and `hook hints` (optional). There is no return value.
+> The `finally` hook **MUST** run after the `before`, `after`, and `error` stages. It accepts a `hook context` (required), `evaluation details` (required) and `hook hints` (optional). It has no return value.
```

Comment:
IMO I think we can just go for a break here - but it would almost certainly result in a breaking change in most SDKs. However, as was discussed in the meeting, we haven't yet hardened the `hooks` spec, and practically, we could therefore implement this in the SDK in a breaking way without a major version increase if we decided that was the right path.

I don't think this is a completely unreasonable approach; hooks which implement `finally` are so far pretty rare, partly because they haven't been so useful since we are missing this data.

I'd like to settle this point before merging this PR.

What do others think? cc @nicklasl @askpt @lukas-reining @aepfli @justinabrahms @federicobond @kinyoklion @thomaspoignant

Reply:
Typically, I would avoid a breaking change, but in this case, I think it's okay. As Todd mentioned, it shouldn't impact the majority of users, and the compiler should help avoid any runtime issues. It also makes the `finally` stage much more useful. Having said that, if we can achieve the same goals in a non-breaking way, please let me know.

Reply:
> I don't think this is a completely unreasonable approach; hooks which implement finally are so far pretty rare, partly because they haven't been so useful since we are missing this data, and the break would be very obvious to the compiler in question for the impacted SDKs

I think this is okay. I mean in the end we could hinder projects that implemented their own hooks to upgrade OF. 
But as you said the change is really small and the probability that there are many hooks using the finally method implemented in some languages are small. And for some languages it is even a non breaking change for the hooks themselves.

Reply:
> It also makes the finally stage much more useful.

Yes and I totally see the value in the change :)

Reply:
If we are going to break, do we want to couple it with: https://github.com/open-feature/spec/pull/273

I re-worked it to be non-breaking, but it is potentially more clear as a breaking change.

Reply:
(Though maybe we want to avoid a more complicated to fix break).

Reply:
I prefer the non-breaking proposal for hook-data.